### DISCUSSION TITLE
Dependencies for self runners?

### DISCUSSION THREAD

**Discussion by iambenmitchell** [upptime/upptime#475] [Q&A]
Hello,

I am planning on running my self-hosted runner inside of a docker using an ubuntu image. Dockers are extremely stripped down and don't even have basic utilities such as `git`  or even `curl`

I have installed the following:

- NodeJS 16
- NPM
- curl
- git
- wget
- zip
- unzip
- sudo
- dotnet

Is there anything I am missing?

**Comment by Aterfax** [comment]
Given the platform weirdness I just reported in https://github.com/upptime/upptime/issues/548 we could do with a good list of project dependencies even if this is not platform specific.

**Reply by AnandChowdhary** [reply]
damn you're right! tbh I have no idea about the dependencies required for self-runners since I only built it with native GitHub Actions in mind and we added self-runners as a quick config PR. Maybe if GitHub has some docs for what the official runners look like, we can probably ask users to duplicate that config?

**Reply by Aterfax** [reply]
There's some specific details here, though it will depend which Github Runner image is in use (Ubuntu?) https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md

**Reply by AnandChowdhary** [reply]
Yes, Ubuntu, so maybe we can link to the Ubuntu README and ask people to make their custom runners have a similar config?

**Reply by ReenigneArcher** [reply]
I have three suggestions/requests.

1. Optionally use the https://github.com/actions/setup-node action?
2. Optional, extra additional commands to run? For example:
3. Optional, container. Linux jobs can use a container image and don't necessarily need to run on the runner host. See https://docs.github.com/en/actions/writing-workflows/choosing-where-your-workflow-runs/running-jobs-in-a-container

```yml
container: <container-name>
setup_node: true
additional_runner_commands: |
  apt-get update
  apt-get install curl ...
  etc.
```

Or, alternatively create a container that has everything needed for these workflows and have the `container` field point to your image.
